### PR TITLE
Advisory Board Experiment

### PR DESCRIPTION
**Pull request by alexbfree**
This commits the basic infrastructure for the experiment. There will be a second PR with the actual UI change for the experimental-ui cohort. This PR also brings in the refactorings to support the Geordi client from the https://github.com/alexbfree/Serengeti/tree/converting-geordi-to-component branch.

### CODE REVIEW

File: app/controllers/animal_selector.coffee [hunk 1]
```diff
@@ -3,9 +3,9 @@ $ = require 'jqueryify'
 template = require 'views/animal_selector'
 FilterMenu = require './filter_menu'
 User = require 'zooniverse/lib/models/user'
-AnalyticsLogger = require 'lib/analytics'
 columnize = require 'lib/columnize'
-ExperimentalSubject = require 'models/experimental_subject'
+Subject = require 'models/subject'
```
Comment (brian-c) [inline app/controllers/animal_selector.coffee]: `Subject` seems to be unused in this file.

File: app/controllers/animal_selector.coffee [hunk 2]
```diff
@@ -72,7 +72,13 @@ class AnimalSelector extends Controller
         targets = @items.filter ':not(".dimmed")'
         targets.first().focus()
         targets.first().click() if targets.length is 1
-        AnalyticsLogger.logEvent 'search', @searchInput.val()
+        Geordi.logEvent {
+          type: 'identify'
```
Comment (brian-c) [inline app/controllers/animal_selector.coffee]: This is logging the same event ("identify") after a search-and-select as actually making an identification. Is that intended?

File: app/controllers/filter_menu.coffee [hunk 3]
```diff
@@ -2,7 +2,9 @@
 template = require 'views/filter_menu'
 $ = require 'jqueryify'
 translate = require 't7e'
-AnalyticsLogger = require 'lib/analytics'
+Subject = require 'models/subject'
```
Comment (brian-c) [inline app/controllers/filter_menu.coffee]: Unused `Subject`

File: app/controllers/filter_menu.coffee [hunk 4]
```diff
@@ -2,7 +2,9 @@
...
+Geordi = require 'lib/geordi_and_experiments_setup'
+ExperimentServer = Geordi.experimentServerClient
```
Comment (brian-c) [inline app/controllers/filter_menu.coffee]: Unused `ExperimentServer`

File: app/controllers/filter_menu.coffee [hunk 5]
```diff
@@ -52,12 +54,28 @@ class FilterMenu extends Controller
...
     result[@characteristic.id] = id
-    AnalyticsLogger.logEvent id
+    subtype = id
+    Geordi.logEvent {
+      type: 'filter'
+      relatedID: id
+      data: {
+        filterType: id
+      }
+      subjectID: ExperimentalSubject.current?.zooniverseId
```
Comment (brian-c) [inline app/controllers/filter_menu.coffee]: Undefined `ExperimentalSubject`

File: app/controllers/subject_viewer.coffee [hunk 6]
```diff
@@ -150,7 +150,14 @@ class SubjectViewer extends Controller
   #   delete @mouseDown
 
   onClickPlay: ->
-    AnalyticsLogger.logEvent 'play', @classification.id, @classification.subject.zooniverseId
+    Geordi.logEvent {
+      type: 'play'
+      relatedID: @classification.id
```
Comment (brian-c) [inline app/controllers/subject_viewer.coffee]: Relevant throughout this file: `@classification.id` isn't defined until the classification has been saved.

File: app/controllers/animal_selector.coffee (hunk 2 above)
Comment (alexbfree) [inline app/controllers/animal_selector.coffee]: good spot - it's supposed to be "search". Copy and paste error, will fix. Exactly what code review is good for!!

File: app/views/experiment_info.eco [hunk 7]
```diff
@@ -10,4 +9,4 @@
...
\ No newline at end of file
+"><span style="color: #cccccc;"><% if User.current?: %><%= User.current.name %><% else: %>(anonymous)<% end %> (<% if ExperimentServer.currentParticipant? && ExperimentServer.currentParticipant.fallback: %>fallback to control<% else: %><% if ExperimentServer.currentCohort?: %><%= ExperimentServer.currentCohort %><% else: %>no <% end %> cohort<% end %>) likes: </span><span class="value"><% if ExperimentServer.currentParticipant? && ExperimentServer.currentParticipant.most_liked_species?: %><%= ExperimentServer.currentParticipant.most_liked_species.join(', ') %><% else: %>N/A<% end %></span>&nbsp;&nbsp;&gt;&nbsp;&nbsp;<span style="color: #cccccc;">Source: </span><span class="subject-source value"><% if ExperimentServer.sources[@subject.zooniverseId]: %><%= ExperimentServer.sources[@subject.zooniverseId] %><% else: %>N/A<% end %></span></div>
```
Comment (brian-c) [inline app/views/experiment_info.eco]: Lost some formatting here?

Comment (alexbfree) [inline app/views/experiment_info.eco]: Doesn't matter, that file isn't used any more, it's only something I use for testing . will probably delete it in a future commit.

**Comment by alexbfree** [comment]
OK - code review comments addressed, and have now added the UI change for experimental UI cohort participants - a new slogan at the top of the classification screen. Ready to go once PR approved.

**Comment by alexbfree** [comment]
Thanks @brian-c !

**Pull request merged**